func.go

```diff
@@ -977,7 +977,7 @@ func funcIndex2(_, v, x any) any {
 				return &expectedObjectError{v}
 			}
 		}
-	case int, float64, *big.Int:
+	case int, int8, int16, int32, int64, float32, float64, *big.Int:
```

Comment:
This isn't all this file needs. I was tinkering earlier and you might need this in the `toInt()`:

```	case int8:
		return int(x), true // We assume that all platforms are at least 32 bit
	case int16:
		return int(x), true // We assume that all platforms are at least 32 bit
	case int32:
		return int(x), true // We assume that all platforms are at least 32 bit
	case int64:
		if uintSize >= 64 || x <= 2147483647 {
			return int(x), true
		}
		return int(x), false // Overflow

	case uint:
		if x <= 2147483647 || (uintSize == 64 && x <= 9223372036854775807) {
			return int(x), true
		}
		return math.MaxInt, false // Overflow
	case uint8:
		return int(x), true // We assume that all platforms are at least 32 bit
	case uint16:
		return int(x), true // We assume that all platforms are at least 32 bit
	case uint32:
		if uintSize > 32 || x <= 2147483647 {
			return int(x), true
		}
		return math.MaxInt, false // Overflow
	case uint64:
		if uintSize == 64 && x <= 9223372036854775807 {
			return int(x), true
		}
		return math.MaxInt, false // Overflow```

Reply:
Maybe similar in `toFloat` (for math operations)

Reply:
Test case would be to use the `funcHas` or `slice` or `sliceString` functions as they take the `toInt` code path
toFloat is called by things like `funcFlatten` or `Isnan` or some others.

Reply:
After looking at it, that func.go file is a mess with these:

Take a look at `funcAbs`, or `funcLength` for example)

*sigh*. Thoughts?

Reply:
Even worse: `operator.go`... checkout `funcOpAdd` or most of the stuff in there, for example.